usr/src/uts/common/io/vioif/vioif.c

```diff
@@ -887,21 +877,31 @@ vioif_process_rx(struct vioif_softc *sc)
 		sc->sc_ipackets++;
 
 		virtio_free_chain(ve);
-		mac_rx(sc->sc_mac_handle, NULL, mp);
-		i++;
+
+		if (lastmp == NULL) {
+			mphead = mp;
+		} else {
+			lastmp->b_next = mp;
+		}
+		lastmp = mp;
+		num_processed++;
 	}
 
-	return (i);
+	if (mphead != NULL) {
+		mac_rx(sc->sc_mac_handle, NULL, mphead);
+	}
+
+	return (num_processed);
 }
 
-static void
+static int
```

Comment:
Everywhere else that counts were returned, you've used uint_t's, but here it uses an int. Is there any particular reason for that?